geoscilabs/gravity/gravityDike.py

```diff
@@ -0,0 +1,324 @@
+            verticalalignment="top",
+            fontsize=10,
+        )
+    ax0.grid(True)
+    ax0.set_ylabel(r"$\Delta g_z$" + "(mGal)", fontsize=16)
+    ax0.set_xlabel("x (m)", fontsize=16)
+    printGrapha(delta_rho, z1, z2, b, beta, ax1, stationSpacing)
+    printDike(ax2, z1, z2, b, beta, 5)
+    if B:
+        plotdata.append(
+            [
+                copy.deepcopy(Y[:, 0]),
+                copy.deepcopy(respNS),
+                colorList[index],
+                showText,
+                textLocation,
+            ]
+        )
+        index += 1
+        if index > len(colorList) - 1:
+            index = 0
+    plt.tight_layout()
+    plt.show()
+
+
+# draw the figure of the Dike
+def printDike(axeToDraw, z1, z2, b, beta, x):
+    axeToDraw.plot(0, 0, ".")
+    axeToDraw.plot([-6, 6], [0, 0], color="black", linewidth=1.5)
+    x1 = -z1 * float(np.tan(beta))
+    x3 = b - z1 * float(np.tan(beta))
+    x2 = -z2 * float(np.tan(beta))
+    x4 = b - z2 * float(np.tan(beta))
+    x1 = min(6, abs(x1)) * getSigned(x1)
+    x2 = min(6, abs(x2)) * getSigned(x2)
+    x3 = min(6, abs(x3)) * getSigned(x3)
+    x4 = min(6, abs(x4)) * getSigned(x4)
+    axeToDraw.plot([x1, x3], [-z1, -z1], color="black", linewidth=1.0)
+    axeToDraw.plot([x2, x4], [-z2, -z2], color="black", linewidth=1.0)
+    if beta != 0:
+        axeToDraw.plot(
+            [x1, x2],
+            [x1 / np.tan(beta), x2 / np.tan(beta)],
+            color="black",
+            linewidth=1.0,
+        )
+        axeToDraw.plot(
+            [x3, x4],
+            [(x3 - b) / np.tan(beta), (x4 - b) / np.tan(beta)],
+            color="black",
+            linewidth=1.0,
+        )
+    else:
+        axeToDraw.plot([x1, x2], [-z1, -z2], color="black", linewidth=1.0)
+        axeToDraw.plot([x3, x4], [-z1, -z2], color="black", linewidth=1.0)
+    axeToDraw.plot(
+        [0, -z1 * float(np.tan(beta))], [0, -z1], ":", color="black", linewidth=1.0
+    )
+    axeToDraw.plot([-6, 6], [-6, -6], color="white", linewidth=1.0)
+    axeToDraw.plot([-6, 6], [1, 1], color="white", linewidth=1.0)
+    axeToDraw.plot([-6, -6], [-6, 1], color="white", linewidth=1.0)
+    axeToDraw.plot([6, 6], [1, -6], color="white", linewidth=1.0)
+    axeToDraw.set_title("Position of Block", fontsize=16)
+
+
+# Get the sign of the input
+def getSigned(a):
+    if a > 0:
+        return 1
+    else:
+        return -1
+
+
+# generate the data of table
+def datagenerator(delta_rho, z1, z2, b, beta, stationSpacing):
+    respEW = 0
+    gravity_change = []
+    xmax = 6.0
+    npts = int(1 / stationSpacing)
+    x = np.linspace(-xmax, xmax, num=npts)
+    y = x.copy()
+    X, Y = np.meshgrid(x, y)
+    for each in range(npts):
+        gravity_change.append(calculategravity(delta_rho, z1, z2, b, beta, x[each]))
+    return respEW, gravity_change, X, Y
+
+
+# get the value of the delta gravity
+def calculategravity(delta_rho, z1, z2, b, beta, x) -> float:
+    G = 6.67259e-8

```

Comment:
I think it would be better here to add in the import:
`from scipy.constants import G` (be careful with the units)